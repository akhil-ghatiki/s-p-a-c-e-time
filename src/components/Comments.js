import React from 'react';
import { makeStyles } from "@material-ui/core/styles";
import { Avatar, Box, Link, Typography } from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
  grow: {
    flexGrow: 1,
  },
  comment: {
    color: '#505050',
    fontFamily: '-apple-system, BlinkMacSystemFont,'
      + ' \'Segoe UI\', \'Roboto\', \'Oxygen\','
      + ' \'Ubuntu\', \'Cantarell\', \'Fira Sans\', \'Droid Sans\','
      + ' \'Helvetica Neue\', sans-serif',
  },
  inline: {
    color: '#204060',
    flexGrow: 1,
    display: 'inline',
    marginRight: 10,
  },
  lowText: {
    flexGrow: 1,
    display: 'inline',
    marginRight: 10,
    color: '#808080',
    fontSize: 12
  }
}));

export default function Comments(props) {
  const classes = useStyles();
  if(props.gitIssue !== undefined){
  return (
    <div>
      <Box display="flex" flexDirection="row" p={1}>
        <Box p={1}>
          <Avatar src={props.commentData.ImgSrc}></Avatar>
        </Box>
        <Box p={1}>
          <Typography className={classes.inline}>{props.commentData.userName}</Typography><br/>
          <Typography className={classes.lowText}>{props.commentData.createdDateTime}</Typography>
          <Link target="_blank" rel="noopener noreferrer" href={props.commentData.commentUrl} color="inherit" className={classes.lowText}>#{props.commentData.id}</Link>
          <Typography className={classes.comment}>{props.commentData.comment}</Typography>
        </Box>
      </Box>
    </div>
  );
  }
  else
  return(<div></div>);
}
