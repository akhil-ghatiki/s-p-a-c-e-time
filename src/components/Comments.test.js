import React from 'react';
import {render} from '@testing-library/react';
import expect from 'expect';
import Comments from './Comments';
import {configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

// test('renders rage text', () => {
//   const { getByText } = render(<App />);
//   const textElement = getByText(/Rage, rage against the dying of the light/i);
//   expect(textElement).toBeInTheDocument();
// });

describe('Comments', () => {
  it('should render correctly', () => {
    const comment = {};
    const component = render(<Comments commentData={comment}/>);
    expect(component).toMatchSnapshot();
  });
});