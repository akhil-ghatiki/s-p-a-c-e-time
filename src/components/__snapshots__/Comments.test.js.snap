// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="MuiBox-root MuiBox-root-5"
        >
          <div
            class="MuiBox-root MuiBox-root-6"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiAvatar-fallback"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiBox-root MuiBox-root-7"
          >
            <p
              class="MuiTypography-root makeStyles-inline-3 MuiTypography-body1"
            />
            <p
              class="MuiTypography-root makeStyles-lowText-4 MuiTypography-body1"
            />
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-lowText-4 MuiTypography-colorInherit"
              rel="noopener noreferrer"
              target="_blank"
            >
              #
            </a>
            <p
              class="MuiTypography-root makeStyles-comment-2 MuiTypography-body1"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="MuiBox-root MuiBox-root-5"
      >
        <div
          class="MuiBox-root MuiBox-root-6"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiAvatar-fallback"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-7"
        >
          <p
            class="MuiTypography-root makeStyles-inline-3 MuiTypography-body1"
          />
          <p
            class="MuiTypography-root makeStyles-lowText-4 MuiTypography-body1"
          />
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-lowText-4 MuiTypography-colorInherit"
            rel="noopener noreferrer"
            target="_blank"
          >
            #
          </a>
          <p
            class="MuiTypography-root makeStyles-comment-2 MuiTypography-body1"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
